### ISSUE TITLE
Better "Documentation" link from pub package page

### ISSUE THREAD

**Issue by srawlins** [dart-lang/pub-dartlang-dart#1174]
Here's the current link text for documentation:

![screen shot 2018-04-05 at 5 40 12 pm](https://user-images.githubusercontent.com/103167/38398436-718d4492-38f8-11e8-8739-39a2a38e9864.png)

The link text is always: "pub.dartlang.org/documentation/<first letter of package>...". This is not very useful.

Maybe just link the word "Documentation"? Maybe not make bold? Maybe link to "/documentation/<packag-name>"?

**Comment by isoos** [comment]
@mit-mit

**Comment by mit-mit** [comment]
As discussed, let try to pull things up like this:

**About**
Flutter plugin for displaying inline video with other Flutter widgets on Android and iOS.

[Homepage](http://github.com/flutter/plugins/tree/master/packages/video_player), [API Docs](http://pub.dartlang.org/documentation/video_player/0.4.0/)

**Comment by jakemac53** [comment]
How about moving this to a Tab instead? This is the primary thing I come to the pub site for now, but it always takes me a second to remember where to find it. 

![screenshot from 2018-04-12 12-05-03](https://user-images.githubusercontent.com/984921/38698408-c11820a8-3e49-11e8-9c23-0252e821825f.png)

I think this is a lot more discoverable and its worth surfacing as a first class item for the package.

**Comment by isoos** [comment]
Making it tabs has been discussed and postponed for now, because it requires to solve a couple of design problems, a few examples:

- What should happen with the dartdoc-specific search? How to integrate it with the site search, filter only on the currently selected package version?

- How shall we handle the design of the dartdoc-content, which is assuming to own the entire page? Slapping the tab bar on top of it may be easy, but the header and tabs contain a few dynamic information (analysis score, platform tags, latest versions), and visually seamless integration requires some design work.

Until these happen, we'll have #1194 as an interim solution. However, would you mind to open a new issue with the screenshot? I'd like to have it tracked here too (but the current one will be closed once the PR lands).

**Comment by jakemac53** [comment]
> What should happen with the dartdoc-specific search? How to integrate it with the site search, filter only on the currently selected package version?

Oh - so I guess this isn't intuitive, but I was actually envisioning this just being a link to the same page that exists today. Not an actual `tab`. Maybe some extra UI could indicate that while still leaving it in the prominent position at the top?

**Comment by jakemac53** [comment]
Something like this?

![button](https://user-images.githubusercontent.com/984921/38700282-60c8c88c-3e4f-11e8-942a-95134d47a9a9.png)

**Comment by isoos** [comment]
Yeah, but that doesn't solve the UX-related issue - the inconsistent content flow: the other tabs stay on the same page, showing different content, while this would navigate to a different URL. Furthermore we still need to match the design there to remain consistent.

Again: we want to do it eventually, but not right now.

**Comment by jakemac53** [comment]
> Yeah, but that doesn't solve the UX-related issue - the inconsistent content flow: the other tabs stay on the same page, showing different content, while this would navigate to a different URL

Ya - I tried to highlight that its not a tab but its hard because the tabs don't have borders or background colors to distinguish them as tabs (if they did this could more much more easily be differentiated). I think they would look a bit better that way anyways but thats just me ;).

> Furthermore we still need to match the design there to remain consistent.

I don't think this is an issue as long as its clear its a link and not a tab.

**Comment by mit-mit** [comment]
We could also consider pulling it up into the links right below the package name?

![screen shot 2018-04-13 at 10 32 38](https://user-images.githubusercontent.com/13644170/38725044-5539c5ae-3f06-11e8-8ac6-5d5da532b738.png)

**Comment by jakemac53** [comment]
The issue for me is it seems to be most similar to the things that are tabs. It just doesn't seem to make sense anywhere else to me personally.

**Comment by isoos** [comment]
https://github.com/dart-lang/pub-dartlang-dart/issues/1214 makes this a bit more complicated:
- we have our generated content,
- the package may specify its URL for documentation, which may be (a) 404, (b) a generic homepage, or (c) a dartdoc-generated content.

I think we should display both the pub-generated API docs, and the documentation link.

**Comment by isoos** [comment]
Putting the pubspec-provided Documentation link after the Homepage is not looking as good as I would have expected:

<img width="257" alt="screen shot 2018-04-18 at 10 32 25" src="https://user-images.githubusercontent.com/4778111/38920600-e992cc28-42f3-11e8-9eee-813d905af9a9.png">

**Comment by isoos** [comment]
We've deployed a new version of the homepage and documentation link handling, which also includes the pubspec-provided documentation link.

**Comment by srawlins** [comment]
Yeees yes yes yes **yes** I love it! I'm so excited about everything pub! It's so beautiful now, thanks @isoos !!
